catalyst-gateway/bin/src/service/common/auth/rbac/token.rs

```diff
@@ -113,11 +106,12 @@ impl CatalystRBACTokenV1 {
         if catalyst_id.nonce().is_none() {
             return Err(anyhow!("Catalyst ID must have nonce"));
         }
-
-        if REGEX.is_match(token) {
-            return Err(anyhow!(
-                "Catalyst ID mustn't have role or rotation specified"
-            ));
+        let (role, rotation) = catalyst_id.role_and_rotation();
```

Comment:
No, this is not correct and these changes are not required for this PR, 
revert these changes and leave using `REGEX`